handlers/utils/compute_utils.go

```diff
@@ -0,0 +1,1186 @@
+			return types.ContainerJSON{}, err
+		}
+		logrus.Infof("container with id [%s] inspected", result.ID)
+		return inspectResp, nil
+	}
+	return types.ContainerJSON{}, fmt.Errorf("container with id [%v] not found", containerID)
+}
+
+func nameFilter(name string, container *types.Container) bool {
+	names := container.Names
+	if names == nil || len(names) == 0 {
+		return false
+	}
+	found := false
+	for _, n := range names {
+		if strings.HasSuffix(n, name) {
+			found = true
+			break
+		}
+	}
+	return found
+}
+
+func IsInstanceRemoved(instance *model.Instance) bool {
+	client := docker.GetClient(DefaultVersion)
+	container := GetContainer(client, instance, false)
+	return container == nil
+}
+
+func DoInstanceRemove(instance *model.Instance, progress *progress.Progress) error {
+	client := docker.GetClient(DefaultVersion)
+	container := GetContainer(client, instance, false)
+	if container == nil {
+		return errors.New("container not found")
+	}
+	return removeContainer(client, container.ID)
+}
+
+func PurgeState(instance *model.Instance) {
+	client := docker.GetClient(DefaultVersion)
+	container := GetContainer(client, instance, false)
+	if container == nil {
+		return
+	}
+	dockerID := container.ID
+	contDir := ContainerStateDir()
+	files := []string{path.Join(contDir, "tmp-"+dockerID), path.Join(contDir, dockerID)}
+	for _, f := range files {
+		if _, err := os.Stat(f); err == nil {
+			os.Remove(f)
+		}
+	}
+}
+
+func getInstanceHostMapData(event *revents.Event) map[string]interface{} {
+	instance, _ := GetInstanceAndHost(event)
+	client := docker.GetClient(DefaultVersion)
+	var inspect types.ContainerJSON
+	container := GetContainer(client, instance, false)
+	dockerPorts := []string{}
+	dockerIP := ""
+	dockerMounts := []types.MountPoint{}
+	if container != nil {
+		inspect, _ = client.ContainerInspect(context.Background(), container.ID)
+		dockerMounts = getMountData(container.ID)
+		dockerIP = inspect.NetworkSettings.IPAddress
+		if container.Ports != nil && len(container.Ports) > 0 {
+			for _, port := range container.Ports {
+				privatePort := fmt.Sprintf("%v/%v", port.PrivatePort, port.Type)
+				portSpec := privatePort
+				bindAddr := ""
+				if port.IP != "" {
+					bindAddr = fmt.Sprintf("%s:", port.IP)
+				}
+				publicPort := ""
+				if port.PublicPort > 0 {
+					publicPort = fmt.Sprintf("%v:", port.PublicPort)
+				} else if port.IP != "" {
+					publicPort = ":"
+				}
+				portSpec = bindAddr + publicPort + portSpec
+				dockerPorts = append(dockerPorts, portSpec)
+			}
+		}
+	}
+	update := map[string]interface{}{
+		"instance": map[string]interface{}{
+			"+data": map[string]interface{}{
+				"dockerContainer": container,
+				"dockerInspect":   inspect,
+				"+fields": map[string]interface{}{
+					"dockerHostIp": DockerHostIP(),
+					"dockerPorts":  dockerPorts,
+					"dockerIp":     dockerIP,
+				},
+			},
+		},
+	}
+	if container != nil {
+		update["instance"].(map[string]interface{})["externalId"] = container.ID
```

Comment:
As opposed to this ugly type assertion, you could either A) define the update maps to have the type `map[string]map[string]interface` or define the inner map as a separate variable that you could reference directly here.